src/Feersum.CompilerServices/Compile/Compiler.fs

```diff
@@ -26,10 +47,9 @@ type EmitCtx =
       ScopePrefix: string
       EmitSymbols: bool
       Exports: Map<string, string>
-      EnvSize: int option
       Externs: Map<string, TypeDefinition>
-      mutable Environment: VariableDefinition option
-      ParentEnvSize: int option
+      ParentEnvironment: EnvInfo option
```

Comment:
Ideally we wouldn’t _care_ about the parent environment if we don/t use it. If we used it then the parent would be accessible through _our_ environment. On the fence here as we do ‘fast track’ captured lookups by starting at the parent environment.

Reply:
We may get rid of the vestigial environments, and the need to know the parent like this, if we thread empty environments rather than creating new types for them.